### DISCUSSION TITLE
Pattern question on who to set as object owner

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#174] [Questions] [answered]
### Describe your question in detail.

When we create an object to store some data for each user. we have two options for object owner. It can be either an app signer or user him/herself. Which one do you guys recommend to use?

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

N/A

### Which SDK or tool are you using? (if any)

N/A

**Comment by kshitijc1506** [comment, marked as answer]
Depending on what features you want, I think ultimately both should be fine right - if it's stored under the user's account, then we just need to ensure it's [Soulbound](https://github.com/aptos-labs/aptos-core/blob/14cfd2da6e3898e82b87607451fd925daa4e5655/aptos-move/framework/aptos-framework/sources/object.move#L428)/unique. 

If stored using the app signer (i.e. resource account for a more decentralized approach), there _might_ be less flexibility in that users can't directly manipulate their own state (using their signer) through contract calls. Is there any reason you think this could be useful for certain types of applications? To me it seems like a reasonable feature for certain DeFi apps. 

For greater flexibility, we need to store the [ExtendRef](https://aptos.dev/standards/aptos-object/#object-capabilities-refs) under the app/protocol account be able to add more data/resources to the object. As long as the app/admin has access to the Object refs, I don't really see why storing using app signer is necessary in the context of a social app.

**Reply by 0x-j** [reply]
Good point, I'll place it under each user him/herself now. It's good enough for my simple friend tech clone hehe.
